src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/EnabledProductInProductAssociationItemExtension.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Bundle\ApiBundle\Doctrine\QueryItemExtension;
+
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Extension\QueryItemExtensionInterface;
+use ApiPlatform\Core\Bridge\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use Doctrine\ORM\QueryBuilder;
+use Sylius\Bundle\ApiBundle\Context\UserContextInterface;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Sylius\Component\Product\Model\ProductAssociationInterface;
+
+/** @experimental */
+final class EnabledProductInProductAssociationItemExtension implements QueryItemExtensionInterface
+{
+    public function __construct(
+        private ChannelContextInterface $channelContext,
```

Comment:
I'm not sure if this is also the case, but [here](https://github.com/Sylius/Sylius/pull/15023/files#diff-8959ea496540a9ab65c3bee64c4e431703e5d512068389bab567f3524dae9cccL41-R59) we got rid of the channel context in favor of `$context[ContextKeys::CHANNEL]`